catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SendEvent.java

```diff
@@ -0,0 +1,234 @@
+
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.event.DeliveryMethod;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.QueryResponseImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.QueryResponseTransformer;
+import net.opengis.cat.csw.v_2_0_2.ElementSetType;
+import net.opengis.cat.csw.v_2_0_2.GetRecordsType;
+import net.opengis.cat.csw.v_2_0_2.QueryType;
+import net.opengis.cat.csw.v_2_0_2.ResultType;
+
+public class SendEvent implements DeliveryMethod {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SendEvent.class);
+
+    TransformerManager transformerManager;
+
+    URI callbackUrl;
+
+    String outputSchema;
+
+    ElementSetType elementSetType;
+
+    List<QName> elementName;
+
+    String mimeType;
+
+    GetRecordsType request;
+
+    ResultType resultType;
+
+    QueryRequest query;
+
+    public SendEvent(TransformerManager transformerManager, URI callbackUrl, GetRecordsType request,
+            QueryRequest query) {
+        this.transformerManager = transformerManager;
+        this.query = query;
+        this.callbackUrl = callbackUrl;
+        this.request = request;
+        this.outputSchema = request.getOutputSchema();
+        this.mimeType = request.getOutputFormat();
+        QueryType queryType = (QueryType) request.getAbstractQuery()
+                .getValue();
+        this.elementName = queryType.getElementName();
+        this.elementSetType = (queryType.getElementSetName() != null) ?
+                queryType.getElementSetName()
+                        .getValue() :
+                null;
+        this.resultType =
+                request.getResultType() == null ? ResultType.HITS : request.getResultType();
+    }
+
+    private String sendEvent(Operation operation, Metacard... metacards) {
+        try (CloseableHttpClient httpClient = getClosableHttpClient()) {
+            List<Result> results = new ArrayList<>(metacards.length);
+            for (int i = 0; i < metacards.length; i++) {
+                results.add(new ResultImpl(metacards[i]));
+            }
+
+            SourceResponse queryResponse = new QueryResponseImpl(query,
+                    results,
+                    true,
+                    metacards.length);
+
+            LOGGER.debug("Attempting to transform an Event with mime-type: {} & outputSchema: {}",
+                    mimeType,
+                    outputSchema);
+            Map<String, Serializable> arguments = new HashMap<>();
+            if (elementName != null) {
+                arguments.put(CswConstants.ELEMENT_NAMES, elementName.toArray());
+            }
+
+            BinaryContent binaryContent = null;
+            if (CswConstants.CSW_OUTPUT_SCHEMA.equals(outputSchema)) {
```

Comment:
If you invert this condition you can reduce cyclomatic complexity of this method.